base/common/DateLUTImpl.cpp

```diff
@@ -157,3 +160,80 @@ DateLUTImpl::DateLUTImpl(const std::string & time_zone_)
         years_months_lut[year_months_lut_index] = first_day_of_last_month;
     }
 }
+
+
+#if !defined(ARCADIA_BUILD) /// Arcadia's variant of CCTZ already has the same implementation.
+
+/// Prefer to load timezones from blobs linked to the binary.
+/// The blobs are provided by "tzdata" library.
+/// This allows to avoid dependency on system tzdata.
+namespace cctz_extension
+{
+    namespace
+    {
+        class Source : public cctz::ZoneInfoSource
+        {
+        public:
+            Source(const char * data_, size_t size_) : data(data_), size(size_) {}
+
+            size_t Read(void * buf, size_t bytes) override
+            {
+                if (bytes > size)
+                    bytes = size;
+                memcpy(buf, data, bytes);
+                data += bytes;
+                size -= bytes;
+                return bytes;
+            }
+
+            int Skip(size_t offset) override
+            {
+                if (offset <= size)
+                {
+                    data += offset;
+                    size -= offset;
+                    return 0;
+                }
+                else
+                {
+                    errno = EINVAL;
+                    return -1;
+                }
+            }
+        private:
+            const char * data;
+            size_t size;
+        };
+
+        std::unique_ptr<cctz::ZoneInfoSource> custom_factory(
+            const std::string & name,
+            const std::function<std::unique_ptr<cctz::ZoneInfoSource>(const std::string & name)> & fallback)
+        {
+            std::string name_replaced = name;
+            std::replace(name_replaced.begin(), name_replaced.end(), '/', '_');
+            std::replace(name_replaced.begin(), name_replaced.end(), '-', '_');
+
+            /// These are the names that are generated by "ld -r -b binary"
+            std::string symbol_name_data = "_binary_" + name_replaced + "_start";
+            std::string symbol_name_size = "_binary_" + name_replaced + "_size";
+
+            const void * sym_data = dlsym(RTLD_DEFAULT, symbol_name_data.c_str());
```

Comment:
I haven't found any function for this, but maybe it's possible to get a symbol address without `dlsym`? It should be possible to lookup symbols in static ELF binaries.

Reply:
Yes, it's also possible: we can open ELF file and navigate its sections, see ELF.h/cpp.
But I don't know any convenient methods for it.

See also: https://stackoverflow.com/questions/34888385/dlsym-like-functionality-for-non-dynamically-loaded-code

Reply:
Less hacky way is to declare every needed symbol as `extern const char *` variable and access directly. But we access them by their names. Nevertheless, we can create a file with index of these symbols at build time and compile it.